### PR TITLE
File storage

### PR DESCRIPTION
**Pull request by MatthewKirik**
Added more tests and fixed functionality.

### CODE REVIEW

File: FileStorage.Tests/FileListTests.cs [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+        //Assert
+        var editedMockObjects = MockObjects
+            .Skip(1)
+            .Append(new MockDto {Id = 12, Text = "Hello edited"});
+        string jsonAfterRemove = GetJsonLines(editedMockObjects);
+        string fileContent = fileSystem.File.ReadAllText(_fileName);
+        fileContent.Should().Be(jsonAfterRemove);
+    }
+}
```
Comment (gurug-prog) [inline FileStorage.Tests/FileListTests.cs]: add \n at EOF

File: FileStorage/FileList.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    public void EditFirst(Func<T, bool> predicate, Func<T, T> map)
+    {
+        var removed = Remove(predicate, 1)
+            .FirstOrDefault();
+        if (removed == null) return;
+        var edited = map(removed);
+        Add(edited);
+    }
+}
```
Comment (gurug-prog) [inline FileStorage/FileList.cs]: add \n at EOF

File: FileStorage.Tests/FileListTests.cs [hunk 3]
```diff
@@ -0,0 +1,219 @@
+﻿using System.IO.Abstractions.TestingHelpers;
+using FluentAssertions;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace FileStorage.Tests;
+
+public class FileListTests
+{
+    private class MockDto
```
Comment (gurug-prog) [inline FileStorage.Tests/FileListTests.cs]: It's Dummy, not Mock. So rename it as DummyDto.

File: FileStorage.Tests/FileListTests.cs [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    {
+        public int Id { get; set; }
+        public string Text { get; set; }
```
Comment (gurug-prog) [inline FileStorage.Tests/FileListTests.cs]: Consider fields as nullable types cause ur tests have warnings bout it.

**Review by gurug-prog — CHANGES_REQUESTED** [review]
Good tests bro. However, there are some warnigns you've to resolve.

File: FileStorage.Tests/FileListTests.cs (hunk 4 above)
Comment (MatthewKirik) [reply FileStorage.Tests/FileListTests.cs]: I think I will add default values instead

**Review by gurug-prog — APPROVED** [review]
LGTM!

**Pull request merged**